### PR TITLE
[REVIEW] Matching get_dummies & select_dtypes behavior to pandas

### PR DESCRIPTION
**Pull request by galipremsagar**
This fixes: #2203 , #1853 

Changes made in this PR:

- [x] Remove `cudf.DataFrame.get_dummies` as we have `cudf.get_dummies` and dask has changes merged which will work with `cudf.get_dummies`
- [x] fix issues in `select_dtypes`
- [x] fix index not being set in `select_dtypes` (inconsistency with pandas)
- [x] changle default type of get_dummies from float64 to uint8
- [x] allow lists & dicts for `prefix_sep`
- [x] add support for fallback dtypes to be encoded if none are given for encoding
- [x] add & revamp tests for `get_dummies`
- [x] add tests for `select_dtypes`

cc: @randerzander , @beckernick @kkraus14

### CODE REVIEW

**Comment by galipremsagar** [comment]
This is ready for review.

File: python/cudf/cudf/reshape/general.py [hunk 1]
```diff
@@ -171,7 +171,7 @@ def get_dummies(
     cats={},
     sparse=False,
     drop_first=False,
-    dtype="float64",
+    dtype="uint8",
```
Comment (kkraus14) [inline python/cudf/cudf/reshape/general.py]: We don't really support uint8 yet, we should use `int8` instead.

Comment (kkraus14) [reply python/cudf/cudf/reshape/general.py]:
```suggestion
    dtype="int8",
```

File: python/cudf/cudf/reshape/general.py [hunk 2]
```diff
@@ -202,7 +202,7 @@ def get_dummies(
         columns. Note this is different from pandas default behavior, which
         encodes all columns with dtype object or categorical
     dtype : str, optional
-        output dtype, default 'float64'
+        output dtype, default 'uint8'
```
Comment (kkraus14) [inline python/cudf/cudf/reshape/general.py]:
```suggestion
        output dtype, default 'int8'
```

**Comment by kkraus14** [comment]
@galipremsagar you need to add a `check_dtype=False` to the `assert_eq` calls where we return int8 vs uint8

**Comment by galipremsagar** [comment]
> @galipremsagar you need to add a `check_dtype=False` to the `assert_eq` calls where we return int8 vs uint8

Added this now Keith.

**Pull request merged**